ExampleMod/Items/Weapons/ExampleYoyo.cs

```diff
@@ -38,6 +39,25 @@ public class ExampleYoyo : ModItem
 			item.shoot = ProjectileType<ExampleYoyoProjectile>();
 		}
 
+		// Make sure that your item can even receive these prefixes (check the vanilla wiki on prefixes)
+		// These are the ones that reduce damage of a melee weapon
+		private static readonly int[] unwantedPrefixes = new int[] { PrefixID.Terrible, PrefixID.Dull, PrefixID.Shameful, PrefixID.Annoying, PrefixID.Broken, PrefixID.Damaged, PrefixID.Shoddy};
+
+		public override bool AllowPrefix(int pre) {
+			// return false to make the game reroll the prefix
+
+			// DON'T DO THIS BY ITSELF:
+			// return false;
+			// This will get the game stuck because it will try to reroll every time. Instead, make it have a chance to return true
+
+			if (Array.IndexOf(unwantedPrefixes, pre) > -1) {
```

Comment:
`return Array.,IndexOf(unwantedPrefixes, pre) == -1`
With comment "return true (allow prefix) if not in the array of unwantedPrefixes (IndexOf returns -1 if item not found)"

Reply:
I usually don't like "compound" statements like these in a "showcase" setting, the logic is less easy to follow. I'll just explain the method more instead

Reply:
plus, IndexOf doesn't always return -1 when not found :p 
`Because most arrays have a lower bound of zero, this method generally returns -1 if value isn't found. In the rare case that the lower bound of the array is equal to Int32.MinValue(0x80000000) and value isn't found, this method returns Int32.MaxValue (0x7FFFFFFF).`